src/probnum/diffeq/odefiltsmooth/initialize/_initialize.py

```diff
@@ -0,0 +1,45 @@
+"""Interface for ODE filter initialization."""
+
+import abc
+
+from probnum import problems, randprocs, randvars
+
+
+class InitializationRoutine(abc.ABC):
```

Comment:
I opted for making this a class for multiple reasons:
1) the routine should only have the interface (ivp, prior) -> rv. Runge-Kutta initialisation has some additional parameters, which should not be part of the call signature IMO
2) For different ODEs, these routines might offer multiple options
3) it is easier to document some properties, e.g. whether the initialization is exact or not
4) it is a MUCH easier type in the constructor of GaussianIVPFilter